src/CoreFoundation/CFProxySupport.cs

```diff
@@ -602,19 +609,27 @@ public static CFRunLoopSource ExecuteProxyAutoConfigurationScript (NSString prox
 
 			if (resultCallback == null)
 				throw new ArgumentNullException ("resultCallback");
-
-			if (clientContext == null)
-				throw new ArgumentNullException ("clientContext");
-
-			IntPtr source = CFNetworkExecuteProxyAutoConfigurationScript (proxyAutoConfigurationScript.Handle, targetURL.Handle, resultCallback, clientContext);
-			return (source == IntPtr.Zero) ? null : new CFRunLoopSource (source);
+			unsafe {
+				BlockLiteral block_handler = new BlockLiteral ();
+				BlockLiteral *block_ptr_handler = &block_handler;
+				block_handler.SetupBlockUnsafe (static_AutoConfigurationResult, resultCallback);
+		
+				try {
+					IntPtr clientPtr = Marshal.AllocHGlobal (Marshal.SizeOf (clientContext));
```

Comment:
Where is this memory freed? Seems like you can pass the `clientContext` variable directly to the P/Invoke if you make it a ref parameter:

```csharp
extern static /* CFRunLoopSourceRef __nonnull */ IntPtr CFNetworkExecuteProxyAutoConfigurationScript (
				/* CFStringRef __nonnull */ IntPtr proxyAutoConfigurationScript,
				/* CFURLRef __nonnull */ IntPtr targetURL,
				/* CFProxyAutoConfigurationResultCallback __nonnull */ ref BlockLiteral cb,
				/* CFStreamClientContext * __nonnull */ ref CFStreamClientContext clientContext);

// ...

IntPtr source = CFNetworkExecuteProxyAutoConfigurationScript (proxyAutoConfigurationScript.Handle, targetURL.Handle, ref block_handle, ref clientContext);
```